### DISCUSSION TITLE
Can this plugin use the current code and console as context?

### DISCUSSION THREAD

**Discussion by machado-t** [MichelNivard/gptstudio#136] [General]
This feature could be helpful for debugging and as a copilot

**Comment by calderonsamuel** [comment]
Right now that is not the case. This is because it runs as a Rstudio's background job. However, we are planning to work on implementing a version that communicates with the user's global environment to allow for these kinds of requests

**Reply by stevegbrooks** [reply]
is there a timeline for this feature? it would be really great! I'd be happy to help out here, too, but wouldn't know where to start.

**Reply by calderonsamuel** [reply]
Hi, there is no timeline for this. I appreciate your help offer, we basically need a way to communicate between R processes. In this thread you can find some limitations I found when trying to communicate between the background job and the main R session: https://github.com/MichelNivard/gptstudio/issues/133

Feel free to comment any ideas or suggestions. PRs are also welcome 😉

**Reply by stevegbrooks** [reply]
Seems like this is a tough nut to crack - having a background process in RStudio communicate with the environment that spawned it. 

There could be a "good enough" solution here where the dataframes, values, etc... in the users environment is dumped into an RDS file before the shiny app is launched such that the shiny app can read from that RDS file while its spinning up, and then have that environment as context. 

Although it wouldn't be dynamic to changes the user makes during the running of the app, at least the user could pre-plan to an extent and get a lot of value already from that. 

Do you think this could be a viable option?

**Reply by calderonsamuel** [reply]
Could be a viable option as long as we check the size of the objects in the environment before saving. I would prefer to store some sort of metadata instead of the objects directly. We could provide something like `gptstudio::update_memory()` for users to update the context at will

**Reply by stevegbrooks** [reply]
i can work on it and start a PR

**Reply by stevegbrooks** [reply]
See #133 , this will be a better option
